setup.py

```diff
@@ -91,6 +91,7 @@
         "black",
     ],
     "doc": [
+        "Jinja<=3.0.3",
```

Comment:
We can leave the sphinx issue for now.
If the CI passes without the document build, we can merge this PR.

Reply:
OK. Then I think this PR is almost ready. A few minutes to finish the rest tests. However, the sphinx issue needs some more time to get fixed. Or we can just hope sphinx could fix [this issue](https://github.com/sphinx-doc/sphinx/issues/10291) soon.

Reply:
I was expecting it would be fixed soon, but if it does not in the next week, I may ask you or @xinjli to do it.

Reply:
yeah, I can help with this if you want to fix it